modules/core/arrow-extras-data/src/test/kotlin/arrow/data/FingerTreeTest.kt

```diff
@@ -0,0 +1,291 @@
+package arrow.data
+
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Tuple2
+import arrow.data.fingertree.internal.Affix.*
+import arrow.data.fingertree.FingerTree
+import arrow.data.fingertree.FingerTree.*
+import arrow.data.fingertree.internal.Node
+import arrow.data.fingertree.internal.Node.Branch2
+import arrow.data.fingertree.internal.Node.Branch3
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+
+class FingerTreeTest : StringSpec() {
+
+  init {
+
+    /**
+     * prepend()
+     */
+
+    "prepend() should return a single finger tree when the finger tree is empty" {
+      val fingerTree = Empty<Int>()
+
+      fingerTree.prepend(1) shouldBe Single(1)
+    }
+
+    "prepend() should return a deep finger tree with one prefix and one suffix element when the finger tree has only one element" {
+      val fingerTree = Single(1)
+
+      fingerTree.prepend(2) shouldBe Deep(One(2), Empty(), One(1))
+    }
+
+    "prepend() should add the item to the first place of the tree prefix when the finger tree has three prefix elements and an empty deeper tree" {
+      val fingerTree = Deep(Three(1, 2, 3), Empty(), One(4))
+
+      fingerTree.prepend(5) shouldBe Deep(Four(5, 1, 2, 3), Empty(), One(4))
+    }
+
+    "prepend() should create a finger tree with the new affix, the result of the prepend on the deeper tree and the same suffix when the prefix has four elements" {
+      val mockDeeperFingerTree = mockk<FingerTree<Node<Int>>>(relaxed = true)
```

Comment:
Please use property-based testing based off KotlinTest for this. Mocking is not the right approach for these as we're testing behavior and not whether something has been called.